src/RpcServer/RpcServer.cs

```diff
@@ -0,0 +1,243 @@
+            if (authvalues.Length < 2)
+                return false;
+
+            return authvalues[0] == Settings.Default.RpcUser && authvalues[1] == Settings.Default.RpcPass;
+        }
+
+        protected override void Configure()
+        {
+            Settings.Load(GetConfiguration());
+        }
+
+        private static JObject CreateErrorResponse(JObject id, int code, string message, JObject data = null)
+        {
+            JObject response = CreateResponse(id);
+            response["error"] = new JObject();
+            response["error"]["code"] = code;
+            response["error"]["message"] = message;
+            if (data != null)
+                response["error"]["data"] = data;
+            return response;
+        }
+
+        private static JObject CreateResponse(JObject id)
+        {
+            JObject response = new JObject();
+            response["jsonrpc"] = "2.0";
+            response["id"] = id;
+            return response;
+        }
+
+        public override void Dispose()
+        {
+            base.Dispose();
+            if (host != null)
+            {
+                host.Dispose();
+                host = null;
+            }
+        }
+
+        protected override void OnPluginsLoaded()
+        {
+            host = new WebHostBuilder().UseKestrel(options => options.Listen(Settings.Default.BindAddress, Settings.Default.Port, listenOptions =>
+            {
+                if (string.IsNullOrEmpty(Settings.Default.SslCert)) return;
+                listenOptions.UseHttps(Settings.Default.SslCert, Settings.Default.SslCertPassword, httpsConnectionAdapterOptions =>
+                {
+                    if (Settings.Default.TrustedAuthorities is null || Settings.Default.TrustedAuthorities.Length == 0)
+                        return;
+                    httpsConnectionAdapterOptions.ClientCertificateMode = ClientCertificateMode.RequireCertificate;
+                    httpsConnectionAdapterOptions.ClientCertificateValidation = (cert, chain, err) =>
+                    {
+                        if (err != SslPolicyErrors.None)
+                            return false;
+                        X509Certificate2 authority = chain.ChainElements[chain.ChainElements.Count - 1].Certificate;
+                        return Settings.Default.TrustedAuthorities.Contains(authority.Thumbprint);
+                    };
+                });
+            }))
+            .Configure(app =>
+            {
+                app.UseResponseCompression();
+                app.Run(ProcessAsync);
+            })
+            .ConfigureServices(services =>
+            {
+                services.AddResponseCompression(options =>
+                {
+                    // options.EnableForHttps = false;
+                    options.Providers.Add<GzipCompressionProvider>();
+                    options.MimeTypes = ResponseCompressionDefaults.MimeTypes.Append("application/json-rpc");
```

Comment:
Hi Erik, we checked this online and it seems that json-rpc doesn't exist, we should use "application/json" instead.

Reply:
Fixed